addons/mail/static/src/messaging/component/activity/activity_tests.js

```diff
@@ -781,6 +781,103 @@ QUnit.test('activity click on edit', async function (assert) {
     );
 });
 
+QUnit.test('activity edition', async function (assert) {
+    assert.expect(15);
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/activity_format') {
+                assert.step('activity_format');
+                return [{
+                    id: 12,
+                    can_write: true,
+                    icon: 'fa-check',
+                    res_id: 42,
+                    res_model: 'res.partner',
+                }];
+            }
+            return this._super(...arguments);
+        },
+        intercepts: {
+            do_action: function (ev) {
+                assert.step('do_action');
+                assert.strictEqual(
+                    ev.data.action.context.default_res_id,
+                    42,
+                    'Action should have the activity res id as default res id in context');
+                assert.strictEqual(
+                    ev.data.action.context.default_res_model,
+                    'res.partner',
+                    'Action should have the activity res model as default res model in context');
+                assert.strictEqual(
+                    ev.data.action.type,
+                    "ir.actions.act_window",
+                    'Action should be of type "ir.actions.act_window"');
+                assert.strictEqual(
+                    ev.data.action.res_model,
+                    "mail.activity",
+                    'Action should have "mail.activity" as res_model');
+                assert.strictEqual(
+                    ev.data.action.res_id,
+                    12,
+                    'Action should have activity id as res_id');
```

Comment:
reformat so that `)` is at new line, and assert text should have double quote (we need to fix the source to stop copy pasting it wrong)